nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/SdkPackageUpdater.cs

```diff
@@ -254,62 +254,133 @@ private static UpdateResult TryUpdateDependencyVersion(ImmutableArray<ProjectBui
 
         foreach (var buildFile in buildFiles)
         {
-            var updateAttributes = new List<XmlAttributeSyntax>();
-            var packageNodes = FindPackageNode(buildFile, dependencyName);
+            var updateNodes = new List<XmlNodeSyntax>();
```

Comment:
IMO my changes make this method a bit out of control WRT complexity, but when I took a swing at simplifying it it made the PR LESS reviewable instead of more.

I filed an issue on our side to make a separate pass at simplifying things here.